catalog/build/py/package/catalog_build/build.py

```diff
@@ -127,15 +129,175 @@ def get_species_row(taxon_info, taxonomic_group_sets, taxonomic_levels):
+          taxon_rank_map[tax_id] = info["taxonomy"]["rank"]
+        else:
+          print(f"rank not found for tax_id: {tax_id}")
+      
+      # Also extract parent taxa information if available
+      if "classification" in info["taxonomy"]:
+        for rank_level, rank_info in info["taxonomy"]["classification"].items():
+          if isinstance(rank_info, dict) and "id" in rank_info and "name" in rank_info:
+            parent_name = rank_info["name"]
+            parent_id = rank_info["id"]
+            parent_id_str = str(parent_id)
+            if parent_id_str in tax_ids and parent_id_str not in taxon_name_map:
+              taxon_name_map[parent_id_str] = parent_name
+              taxon_rank_map[parent_id_str] = rank_level
+    
+  # Fetch any missing taxa information
+  fetch_taxa_info_in_batches(tax_ids, taxon_name_map, taxon_rank_map, "missing parent taxa")
+
+  named_species_tree = update_species_tree_names(species_tree, taxon_name_map, taxon_rank_map)
+
+  return named_species_tree
+
+
+def fetch_taxa_info_in_batches(tax_ids, taxon_name_map, taxon_rank_map, description="taxa"):
+  """
+  Fetches taxonomic information in batches and updates the provided name and rank maps.
+  
+  Args:
+    tax_ids: List or set of taxonomy IDs to fetch
+    taxon_name_map: Dictionary to update with taxon ID to name mappings
+    taxon_rank_map: Dictionary to update with taxon ID to rank mappings
+    description: Description of the taxa being fetched for logging
+    
+  Returns:
+    None (updates the provided maps in-place)
+  """
+  # Filter out tax_ids that are already in the map
+  missing_tax_ids = [tid for tid in tax_ids if tid not in taxon_name_map and tid != "1"]
+  
+  if not missing_tax_ids:
+    return
+    
+  print(f"Fetching information for {len(missing_tax_ids)} {description}")
+  
+  # Process in batches of 100 to avoid API limitations
+  batch_size = 100
+  for i in range(0, len(missing_tax_ids), batch_size):
+    batch = missing_tax_ids[i:i+batch_size]
+    print(f"Fetching batch {i//batch_size + 1} of {(len(missing_tax_ids) + batch_size - 1)//batch_size} ({len(batch)} taxa)")
+    
+    taxa_info = get_paginated_ncbi_results(
+      f"https://api.ncbi.nlm.nih.gov/datasets/v2/taxonomy/taxon/{','.join(batch)}/dataset_report", 
+      f"{description} batch {i//batch_size + 1}"
+    )
+    
+    for report in taxa_info:
+      tax_id = str(report["taxonomy"]["tax_id"])
+      taxon_name_map[tax_id] = report["taxonomy"]["current_scientific_name"]["name"]
+      if "rank" in report["taxonomy"]:
+        taxon_rank_map[tax_id] = report["taxonomy"]["rank"]
+      else:
+        print(f"rank not found for tax_id: {tax_id}")
+
+
+def ncbi_tree_to_nested_tree(node_id, edges, taxonomy_ids):
+  children = edges.get(str(node_id), {}).get("visible_children", [])
+  children = [str(num) for num in children]
+  # ncbi results odd again, dup children
+  children = set(children)
+  if (len(children) > 0 or node_id in taxonomy_ids):
```

Comment:
```suggestion
  if (len(children) > 0 or int(node_id) in taxonomy_ids):
```

`taxonomy_ids` contains ints